fe2-android/app/src/main/java/com/github/dedis/popstellar/model/network/method/message/data/digitalcash/Transaction.java

```diff
@@ -0,0 +1,130 @@
+package com.github.dedis.popstellar.model.network.method.message.data.digitalcash;
+
+import com.github.dedis.popstellar.utility.security.Hash;
+import com.google.gson.annotations.SerializedName;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
+
+/**
+ * The transaction object
+ */
+public class Transaction {
+  @SerializedName(value = "version")
+  private final int version; // The version of the transaction inputs
+
+  @SerializedName(value = "inputs")
+  private List<Input> inputs; // [Array[Objects]] array of output transactions to use as inputs
+
+  @SerializedName(value = "outputs")
+  private List<Output> outputs; // [Array[Objects]] array of outputs from this transactions
+
+  @SerializedName("lock_time")
+  private final long lockTime; // LockTime
+
+  /**
+   * Transaction constructor
+   * @param version The version of the transaction inputs
+   * @param inputs [Array[Objects]] array of output transactions to use as inputs
+   * @param outputs [Array[Objects]] array of outputs from this transactions
+   * @param lockTime TimeStamp
+   */
+  public Transaction(int version, List<Input> inputs, List<Output> outputs, long lockTime) {
+    this.version = version;
+    this.inputs = Collections.checkedList(inputs, Input.class);
```

Comment:
Why do we need a mutable checked list here? wouldn't it make more sense to simply have an immutable list? This class is immutable, right? (I mean the signatures are mutable but everything else is immutable)

Reply:
Will check later if there are any differences with other parts of the codebase for Object Array in JSON

Reply:
would be removed